src/shared/Microsoft.Git.CredentialManager/Application.cs

```diff
@@ -214,56 +191,41 @@ Task IConfigurableComponent.UnconfigureAsync(ConfigurationTarget target)
-                            // Clear the blank entry
-                            config.UnsetAll(helperKey, Constants.RegexPatterns.Empty);
-                        }
+            GitConfigurationLevel configLevel = target == ConfigurationTarget.System
+                    ? GitConfigurationLevel.System
+                    : GitConfigurationLevel.Global;
 
-                        // Clear app entry
-                        config.UnsetAll(helperKey, Regex.Escape(appPath));
-                    }
-                    break;
+            Context.Trace.WriteLine($"Unconfiguring for config level '{configLevel}'.");
 
-                case ConfigurationTarget.System:
-                    // For machine-wide configuration, we are looking for the following to be set in the system config:
-                    //
-                    // [credential]
-                    //     helper = {appPath}
-                    //
-                    // We should remove the {appPath} entry if it exists.
-                    //
-                    Context.Trace.WriteLine("Removing Git credential helper system configuration...");
-                    config = Context.Git.GetConfiguration(GitConfigurationLevel.System);
-                    config.UnsetAll(helperKey, Regex.Escape(appPath));
-                    break;
+            IGitConfiguration config = Context.Git.GetConfiguration(configLevel);
 
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(target), target, "Unknown configuration target.");
+            // We are looking for the following to be set in the config:
+            //
+            // [credential]
+            //     ...                 # any number of helper entries (possibly none)
+            //     helper =            # an empty value to reset/clear any previous entries (if applicable)
+            //     helper = {appPath} # the expected executable value & directly following the empty value
+            //     ...                 # any number of helper entries (possibly none)
+            //
+            // We should remove the {appPath} entry, and any blank entries immediately preceding IFF there are no more entries following.
+            //
+            Context.Trace.WriteLine("Removing Git credential helper configuration...");
+
+            string[] currentValues = config.GetRegex(helperKey, Constants.RegexPatterns.Any).ToArray();
+
+            int appIndex = Array.FindIndex(currentValues, x => Context.FileSystem.IsSamePath(x, appPath));
+            if (appIndex > -1)
+            {
+                // Check for the presence of a blank entry immediately preceding an app entry in the last position
+                if (appIndex > 0 && appIndex == currentValues.Length - 1 &&
+                    string.IsNullOrWhiteSpace(currentValues[appIndex - 1]))
+                {
+                    // Clear the blank entry
+                    config.UnsetAll(helperKey, Constants.RegexPatterns.Empty);
```

Comment:
What if there are _multiple_ empty entries?

Reply:
Not entirely sure what we can do here to identify a specific line in a config file, that matches the same value, using only `git config` commands. Any ideas?